### PR TITLE
TP CreateHolding functionality

### PR DESCRIPTION
**Pull request by boydjohnson**
Only the last 4 commits are from this PR. This PR will be rebased after #35 is merged into master.
To test run `./bin/dev-tools -a` and then `./bin/run_integration_tests blockchain`

### CODE REVIEW

**Review by peterschwarz — APPROVED** [review]

File: processor/marketplace_processor/holding/holding_creation.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            "txn signer {}".format(account.public_key,
+                                   header.signer_public_key))
+
+    asset = state.get_asset(name=create_holding.holding)
+    if not asset:
+        raise InvalidTransaction(
+            "Failed to create Holding, asset {} does not "
+            "exist.".format(create_holding.holding))
+
+    if create_holding.quantity > 0 and \
```
Comment (delventhalz) [inline processor/marketplace_processor/holding/holding_creation.py]: Quantity could be negative, and non-asset-owners should not be able to create that. The check should be `!= 0`.

File: processor/marketplace_processor/holding/holding_creation.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            "Failed to create Holding, account {} does not exist.".format(
+                create_holding.account))
+
+    if account.public_key != header.signer_public_key:
+        raise InvalidTransaction(
+            "Failed to create Holding, account {} does not belong to "
+            "txn signer {}".format(account.public_key,
+                                   header.signer_public_key))
+
+    asset = state.get_asset(name=create_holding.holding)
```
Comment (delventhalz) [inline processor/marketplace_processor/holding/holding_creation.py]: See typo comment above.

**Review by delventhalz — CHANGES_REQUESTED** [review]

File: processor/marketplace_processor/holding/holding_creation.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        create_holding (CreateHolding): The transaction.
+        header (TransactionHeader): The header of the Transaction.
+        state (MarketplaceState): The wrapper around the context.
+
+    Raises:
+        InvalidTransaction
+            - There is already a Holding with the same identifier.
+            - The account does not exist.
+            - The account holder is the Txn signer.
+            - The asset/holding does not exist.
```
[the remark this answers is not in the document]
Comment (boydjohnson) [reply processor/marketplace_processor/holding/holding_creation.py]: In CreateHolding it is called holding but in Holding it is asset.

File: processor/marketplace_processor/holding/holding_creation.py (hunk 1 above)
Comment (boydjohnson) [reply processor/marketplace_processor/holding/holding_creation.py]: Sounds good.

File: processor/marketplace_processor/holding/holding_creation.py (hunk 2 above)
Comment (boydjohnson) [reply processor/marketplace_processor/holding/holding_creation.py]: I will update the CreateHolding proto. It is assumed the the txn signer is creating the Holding for their account, right?

Comment (delventhalz) [reply processor/marketplace_processor/holding/holding_creation.py]: Yes. They will always be creating holdings for their own account.

**Review by delventhalz — APPROVED** [review]

**Pull request merged**